### PR TITLE
Create web api

### PR DESCRIPTION
**Pull request by VicZhang1**
@DavidZhang1 @SebastianGodelet would you like to take a look this/

Changes are:
1. Create a web api project
2. Handle exceptions
3. APIs(test, version, home...)
4. Dell Single Sign On
5. Https if need

### CODE REVIEW

File: lambda-architecture/LambdaWebAPI/QuestSingleSignOn.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            using (var client = new SSOServiceClient("BasicHttpBinding_ISSOService", "https://externalsso.quest.com/sso/ssoservice.svc"))
+            {
+                var request = new LoginRequestMessage
+                {
+                    UserName = email,
+                    Password = password,
+                    IPAddress = ipAddress
+                };
+                if (ipAddress == "::1")
+                    request.IPAddress = "127.0.0.1";
```
Comment (SebastianGodelet) [inline lambda-architecture/LambdaWebAPI/QuestSingleSignOn.cs]: why switch to IPv4?

File: lambda-architecture/LambdaWebAPI/RequireHttpsAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Net.Http;
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+
+namespace LambdaWebAPI
+{
+    // see http://www.asp.net/web-api/overview/security/working-with-ssl-in-web-api
+    public class RequireHttpsAttribute : AuthorizationFilterAttribute
```
Comment (DavidZhang1) [inline lambda-architecture/LambdaWebAPI/RequireHttpsAttribute.cs]: Why creating a custom RequireHttpsAttribute?
https://msdn.microsoft.com/en-us/library/system.web.mvc.requirehttpsattribute

Comment (VicZhang1) [inline lambda-architecture/LambdaWebAPI/RequireHttpsAttribute.cs]: Done, but this needs to reference System.Web.Mvc.dll which are included in MVC 5.

File: lambda-architecture/LambdaWebAPI/QuestSingleSignOn.cs (hunk 1 above)
Comment (VicZhang1) [inline lambda-architecture/LambdaWebAPI/QuestSingleSignOn.cs]: @SebastianGodelet this is required by SSO service, "::1" is the loopback address in IPv6, if we use this, we will get an exception that:
"Exception in Login method: IP address is invalid. The IP address must be a valid IPv4 address"

**Pull request merged**